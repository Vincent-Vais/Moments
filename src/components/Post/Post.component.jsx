import React, { useState } from "react";
import { Avatar } from "@material-ui/core";

const LIMIT = 15;

function Post({ classes, post, gridClass }) {
  const { username, imageUrl, comment } = post;

  const [slider, setSlider] = useState(false);

  const shortenComment = () =>
    comment.length > LIMIT ? `${comment.slice(0, LIMIT)}...` : comment;

  const handleSlider = (bool) => {
    if (comment.length < LIMIT) return;
    setSlider(bool);
  };

  return (
    <div
      className={`${classes.post} ${gridClass}`}
      onMouseEnter={() => handleSlider(true)}
      onMouseLeave={() => handleSlider(false)}
    >
      <div className={classes.post__header}>
        <Avatar
          className={classes.post__avatar}
          alt={`${username}`}
          src="/static/image1.png"
        />
        <h3 className={classes.post__user}>{username}</h3>
      </div>
      <img
        alt={`${comment.slice(0, 5)}`}
        className={classes.post__img}
        src={imageUrl}
      />
      {slider ? (
        <h4 className={classes.post__slider}>
          <span className={classes.post__bold}>@{username}:</span> {comment}
        </h4>
      ) : (
        <h4 className={classes.post__text}>
          <span className={classes.post__bold}>@{username}:</span>{" "}
          {shortenComment(comment)}
        </h4>
      )}
    </div>
  );
}

export default Post;
