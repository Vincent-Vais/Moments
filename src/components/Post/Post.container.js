import React from "react";
import useStyles from "./Post.styles";

function Container({ Component, ...otherProps }) {
  const classes = useStyles();
  return <Component classes={classes} {...otherProps} />;
}

export default (component) => (props) => (
  <Container Component={component} {...props} />
);
