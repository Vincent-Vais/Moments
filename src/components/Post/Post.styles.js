import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
  post: {
    border: `1px solid ${theme.palette.secondary.main}`,
    borderRadius: "5px",
    background: `${theme.palette.bcg.light}`,
    display: "flex",
    flexDirection: "column",
    height: "70vh",
  },
  post__avatar: {
    height: "5rem",
    width: "5rem",
    fontSize: "3.5rem",
    fontWeight: 500,
    border: `2.5px solid ${theme.palette.primary.main}`,
  },
  post__header: {
    display: "flex",
    alignItems: "center",
    padding: "2rem",
    color: `${theme.palette.text.main}`,
    letterSpacing: "1px",
    flex: "1 1 10%",
  },
  post__img: {
    objectFit: "cover",
    display: "block",
    overflow: "hidden",
    flex: "1 1 80%",
    width: "100%",
    borderBottom: `1px solid ${theme.palette.secondary.main}`,
    borderTop: `1px solid ${theme.palette.secondary.main}`,
  },
  post__user: {
    marginLeft: "1rem",
    fontSize: "2.75rem",
    fontWeight: "400",
  },
  post__text: {
    fontWeight: "300",
    padding: "2rem",
    fontSize: "2.25rem",
    letterSpacing: "1px",
    wordSpacing: "2.15px",
    flex: "1 1 10%",
  },
  post__bold: {
    fontWeight: 700,
    letterSpacing: "1px",
  },
  post__slider: {
    flex: "2 1 80%",
    fontWeight: "300",
    padding: "2rem",
    fontSize: "2.25rem",
    letterSpacing: "1px",
    wordSpacing: "2.15px",
    textAlign: "justify",
  },
}));

export default useStyles;
