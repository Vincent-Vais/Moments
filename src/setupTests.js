import React from "react";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import configureStore from "redux-mock-store";
import { createMuiTheme } from "@material-ui/core";
import { ThemeProvider } from "@material-ui/core/styles";

configure({ adapter: new Adapter() });

export function MockTheme({ children }) {
  const theme = createMuiTheme({});
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}

const middlewares = [];
export const mockStore = configureStore(middlewares);

export const mockClasses = {};
