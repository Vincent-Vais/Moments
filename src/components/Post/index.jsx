import Post from "./Post.component";
import WithData from "./Post.container";

export default WithData(Post);
