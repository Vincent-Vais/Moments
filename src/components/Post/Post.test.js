import React from "react";
import { shallow } from "enzyme";
import Post from "./Post.component";
import { mockClasses } from "../../setupTests";

const mockPost = {
  username: "vince_vais",
  imageUrl:
    "https://images.unsplash.com/photo-1602411583503-195316dad599?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=634&q=80",
  comment:
    "lorem ipsum sit emiet dolor lorem ipsum sit emiet dolorlorem ipsum sit emiet dolorlorem ipsum sitemiet dolor lorem ipsum sit emiet dolorlorem ipsum sit emiet dolor",
};

const getPresentComponent = () => (
  <Post post={mockPost} classes={mockClasses} />
);

describe("Post component", () => {
  it("renders presentational component", () => {
    const component = shallow(getPresentComponent());
    expect(component).toMatchSnapshot();
  });
});
