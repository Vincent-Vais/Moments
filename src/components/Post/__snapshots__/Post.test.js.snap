// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component renders presentational component 1`] = `
<div
  className="undefined undefined"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div>
    <WithStyles(ForwardRef(Avatar))
      alt="vince_vais"
      src="/static/image1.png"
    />
    <h3>
      vince_vais
    </h3>
  </div>
  <img
    alt="lorem"
    src="https://images.unsplash.com/photo-1602411583503-195316dad599?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=634&q=80"
  />
  <h4>
    <span>
      @
      vince_vais
      :
    </span>
     
    lorem ipsum sit...
  </h4>
</div>
`;
